IN THIS MODULE WE ARE GOING TO IMPLEMENT ADA-BOOST 

In [1]:
# To plot all the graph within the note book 
%matplotlib inline

In [2]:
#importing all the required libraries 

import numpy as np
import pandas as pd 
from sklearn import metrics
from sklearn.tree import DecisionTreeClassifier
import seaborn as sns 

In [3]:
#loading the datset 

df=pd.read_csv("credit.csv")
df=pd.DataFrame(df)

In [4]:
#by the above info we can notice that there are more number of non-numerical columns
# converting them into numerical type

for feature in df.columns:
    if df[feature].dtype=='object':
        df[feature]=pd.Categorical(df[feature]).codes

In [8]:
#splitting the datset into training and testing 

train_set=df.head(700)
test_set=df.tail(300)

train_labels=train_set.pop('default')
test_labels=test_set.pop('default')

In [9]:
dt_model=DecisionTreeClassifier(criterion='entropy')

In [11]:
dt_model.fit(train_set,train_labels)

DecisionTreeClassifier(criterion='entropy')

In [12]:
dt_model.score(test_set,test_labels)

0.6633333333333333

Decision tree alone giving accuracy of 66%

In [19]:
from sklearn.ensemble import AdaBoostClassifier
adb=AdaBoostClassifier( n_estimators=50)



In [20]:
adb=adb.fit(train_set,train_labels)

C:\Users\abhis\PycharmProjects\wwk4\venv\Lib\site-packages\sklearn\ensemble\_weight_boosting.py:519: FutureWarning: The SAMME.R algorithm (the default) is deprecated and will be removed in 1.6. Use the SAMME algorithm to circumvent this warning.
  warnings.warn(


In [21]:
y_predict=adb.predict(test_set)

In [22]:
adb.score(test_set,test_labels)

0.7433333333333333

we can observe that accuracy increased from 66% to 74% after using boosting technique 